src/backend/utils/misc/queryenvironment.c

```diff
@@ -494,37 +643,90 @@ static bool _ENR_tuple_operation(Relation catalog_rel, HeapTuple tup, ENRTupleOp
 		switch (catalog_oid) {
 			case RelationRelationId:
 				rel_oid = ((Form_pg_class) GETSTRUCT(tup))->oid;
-				if ((enr = get_ENR_withoid(queryEnv, rel_oid))) {
+				if ((enr = get_ENR_withoid(queryEnv, rel_oid, ENR_TSQL_TEMP))) {
 					list_ptr = &enr->md.cattups[ENR_CATTUP_CLASS];
 					lc = list_head(enr->md.cattups[ENR_CATTUP_CLASS]);
 					ret = true;
 				}
 				break;
+			case DependRelationId:
+				{
+					Form_pg_depend tf1 = (Form_pg_depend) GETSTRUCT((HeapTuple)tup);
+					//if ((enr = find_enr(tf1->classid, tf1->objid, tf1->objsubid))) {
+					if ((enr = find_enr(tf1))) {
```

Comment:
have we done good testing around this code to make sure we are taking care of all dependencies and types properly

Reply:
will add more testcases for this for table variables

Reply:
tests in https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/1440 cover types and dependencies